scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,207 @@
+
+    statuses = {
+        12: (1, "Iteration limit reached."),
+        9: (0, "Optimization terminated successfully."),
+        8: (3, "The problem is unbounded."),
+        7: (2, "The problem is infeasible."),
+        12: (1, "Iteration limit reached."),
+    }
+    # "Iteration limit reached.",
+    # "The problem appears infeasible, as the phase one auxiliary "
+    # "problem terminated successfully with a residual of {0:.1e}, "
+    # "greater than the tolerance {1} required for the solution to "
+    # "be considered feasible. Consider increasing the tolerance to "
+    # "be greater than {0:.1e}. If this tolerance is unnaceptably "
+    # "large, the problem is likely infeasible.",
+    # "The problem is unbounded, as the simplex algorithm found "
+    # "a basic feasible solution from which there is a direction "
+    # "with negative reduced cost in which all decision variables "
+    # "increase.",
+    # "Numerical difficulties encountered; consider trying "
+    # "method='interior-point'.",
+
+    c, A_ub, b_ub, A_eq, b_eq, bounds, x0 = lp
+
+    lb, ub = bounds.T.copy()                # separate bounds, copy->C-cntgs
+    # highs_wrapper solves LHS <= A*x <= RHS, not equality constraints
+    lhs_ub = -np.ones_like(b_ub)*np.inf     # LHS of UB constraints is -inf
+    rhs_ub = b_ub                           # RHS of UB constraints is b_ub
+    lhs_eq = b_eq                           # Equality constaint is inequality
+    rhs_eq = b_eq                           # constraint with LHS=RHS
+    lhs = np.concatenate((lhs_ub, lhs_eq))
+    rhs = np.concatenate((rhs_ub, rhs_eq))
+
+    if issparse(A_ub) or issparse(A_eq):
+        A = vstack((A_ub, A_eq))
+    else:
+        A = np.vstack((A_ub, A_eq))
+    A = csc_matrix(A)
+
+    options = {
+        'presolve': presolve,
+        'sense': 1,  # minimization
+        'solver': solver,
+        'parallel': parallel,
+        'time_limit': time_limit,
+        'message_level': disp * 1,  # 0 is none, 1 is some. Simplex only?
+        'write_solution_to_file': False,
+        'solution_file': 'test.sol',
+        'write_solution_pretty': True,
+    }
+
+    # this doesn't seem to work, and I don't see an equivalent for IPM
+    if maxiter is not None and solver == 'simplex':
+        options['simplex_iteration_limit'] = maxiter
+
+    rhs = _replace_inf(rhs)
+    lhs = _replace_inf(lhs)
+    lb = _replace_inf(lb)
+    ub = _replace_inf(ub)
+
+    res = highs_wrapper(c, A.indptr, A.indices, A.data, lhs, rhs,
+                        lb, ub, options)
+
+    sol = {'x': res.get('x', None),
+           'slack': res.get('slack', None),
+           'fun': res.get('fun', None),
+           'con': res.get('con', None),
+           'status': statuses[res['status']][0],
+           'success': res['status'] == 9,
+           'message': res['message'],
+           'nit': (res.get('simplex_nit', 0) if solver == 'simplex'
+                   else res.get('ipm_nit', 0))
+           }
```

Comment:
We should also return the duals and anything else that's useful for sensitivity analysis.

Reply:
I just checked again, but duals do not appear to be returned with the HighsSolution object, so I'll make an issue asking the HiGHS team about how to do this

Reply:
slack values and lagrange multipliers are now returned by highs_wrapper